zeus/carbon/electricity_maps_carbon_intensity_provider.py

```diff
@@ -0,0 +1,30 @@
+"""Carbon Intensity Provider using ElectrictyMaps API."""
+from zeus.carbon.carbon_intensity_provider import CarbonIntensityProvider
+import requests
+
+
+class ElectricityMapsCarbonIntensityProvider(CarbonIntensityProvider):
+    """Carbon Intensity Provider with ElectricityMaps API."""
+
+    def get_current_carbon_intensity(
+        self, estimate: bool = False, emission_factor_type: str = "direct"
+    ) -> float:
```

Comment:
The signature of the parent abstract function is different from this signature. This defeats the whole purpose of inheritance, because now a downstream class that makes use of `CarbonIntensityProvider` has to distinguish between different implementations to call `get_current_carbon_intensity` correctly.

One solution is to move `estimate` and `emission_factor_type` to `__init__`. I think it's not too bad.

Reply:
ahhhh good catch, did not realize this.